src/test/java/edu/cornell/kfs/concur/services/ConcurAccountValidationServiceTest.java

```diff
@@ -1,484 +1,305 @@
-                MessageFormat.format(concurAccountValidationService.getConfigurationService().getPropertyValueAsString(KFSKeyConstants.ERROR_EXISTENCE), projectCodeErrorMessageString) + KFSConstants.NEWLINE,
-                validationResult.getErrorMessagesAsOneFormattedString());
+
+    private SubObjectCodeService buildMockSubObjectCodeService() {
+        SubObjectCodeService service = Mockito.mock(SubObjectCodeService.class);
+        Mockito.when(service.getByPrimaryIdForCurrentYear(SubObjectCodeEnum.VALID.chart, 
+                SubObjectCodeEnum.VALID.account, SubObjectCodeEnum.VALID.objectCode, SubObjectCodeEnum.VALID.subObjectCode))
+                .thenReturn(SubObjectCodeEnum.VALID.toSubObjectCode());
+        Mockito.when(service.getByPrimaryIdForCurrentYear(SubObjectCodeEnum.INACTIVE.chart,
+                SubObjectCodeEnum.INACTIVE.account, SubObjectCodeEnum.INACTIVE.objectCode,
+                SubObjectCodeEnum.INACTIVE.subObjectCode)).thenReturn(SubObjectCodeEnum.INACTIVE.toSubObjectCode());
+        return service;
     }
-    
-    @Test
-    public void isProjectCodeInactive(){
-        ValidationResult validationResult = concurAccountValidationService.checkProjectCode(ConcurAccountValidationTestConstants.INACTIVE_PROJECT_CODE);
-        String  projectCodeErrorMessageString = ConcurUtils.formatStringForErrorMessage(ConcurConstants.AccountingStringFieldNames.PROJECT_CODE, ConcurAccountValidationTestConstants.INACTIVE_PROJECT_CODE);
-        Assert.assertFalse("Validation was expected to fail but returned true", validationResult.isValid());
-        Assert.assertEquals(
-                "Error message was expected for inactive project code",
-                MessageFormat.format(concurAccountValidationService.getConfigurationService().getPropertyValueAsString(KFSKeyConstants.ERROR_INACTIVE), projectCodeErrorMessageString) + KFSConstants.NEWLINE,
-                validationResult.getErrorMessagesAsOneFormattedString());
+
+    private ProjectCodeService buildMockProjectCodeService() {
+        ProjectCodeService service = Mockito.mock(ProjectCodeService.class);
+        Mockito.when(service.getByPrimaryId(ProjectCodeEnum.VALID.projectCode))
+                .thenReturn(ProjectCodeEnum.VALID.toProjectCode());
+        Mockito.when(service.getByPrimaryId(ProjectCodeEnum.INACTIVE.projectCode))
+                .thenReturn(ProjectCodeEnum.INACTIVE.toProjectCode());
+        return service;
     }
-    
-    @Test
-    public void isAccountInfoValid(){
-        ValidationResult validationResult = concurAccountValidationService.validateConcurAccountInfo(concurAccountInfo);
-        Assert.assertTrue(validationResult.getErrorMessagesAsOneFormattedString(), validationResult.isValid());
+
+    private ProjectCode createProjectCode(String projectCode, boolean active) {
+        ProjectCode project = new ProjectCode();
+        project.setCode(projectCode);
+        project.setActive(active);
+        return project;
     }
```

Comment:
Is this createProjectCode method being used

Reply:
will remove